src/MappingBar/ZoneSearch.tsx

```diff
@@ -1,31 +1,118 @@
-import React, { FC } from 'react';
+import clone from 'lodash/cloneDeep';
+import isEqual from 'lodash/isEqual';
+import startsWith from 'lodash/startsWith';
+import React, { FC, useCallback, useRef, useState } from 'react';
 
 import { TextField } from '@material-ui/core';
 import Autocomplete from '@material-ui/lab/Autocomplete';
+import useEventListener from '@use-it/event-listener';
+
+import { DEFAULT_ZONE } from '../data/constants';
+import { ZoneLight } from '../types';
 
 interface ZoneSearchProps {
-  zoneNames: string[];
+  zoneList: ZoneLight[];
   label: string;
-  value: string | null;
-  update: (zone: string | null) => void;
+  value: ZoneLight;
+  update: (zone: ZoneLight) => void;
 }
+const filterZones = (zoneList: ZoneLight[], state: object) => {
```

Comment:
Explicit return type would be good here.

Reply:
Agreed, but this is actually what they have in their docs. 

https://material-ui.com/api/autocomplete/

![e7ddfe8d1bf89dc32a93b220505c5faa](https://user-images.githubusercontent.com/2829924/95683800-80a4b580-0bbb-11eb-96d4-ca9af28246ea.png)

Reply:
Sure, but you don't pass it to them. You call it in your own code.

Reply:
I can give it a shot. I just copied this because I didn't want to fight TS on this one, which sometimes happens. I'll let you know if it gives me a battle.